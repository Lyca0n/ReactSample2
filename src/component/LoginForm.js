import React from 'react';

export default class LoginForm extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            username: '',
            password: '',
            error:''
        };
    }

    onUsernameChange = (e) => {
        const username = e.target.value;
        this.setState(() => ({ username }));
    };
    onPasswordChange = (e) => {
        const password = e.target.value;
        this.setState(() => ({ password }));
    };

    onSubmit=(e)=>{
        e.preventDefault();
        if(!this.state.username || !this.state.password){
            this.setState(()=>({error : 'Provide descripton and amount'}));
        }else{
            this.setState(()=>({error : ''}));
            this.props.startLogin({});
        }
    };
    render() {
        if (this.state.hasError) {
            // You can render any custom fallback UI
            return <h1>Something went wrong.</h1>;
        } else {
            return (
                <div>
                    <div >{this.state.error && <p>{this.state.error}</p>}</div>
                    <form onSubmit={this.onSubmit}>
                        <input
                            type="text"
                            placeholder="Username"
                            autoFocus
                            value={this.state.username}
                            onChange={this.onUsernameChange}
                        />
                        <input
                            value={this.state.password}
                            onChange={this.onPasswordChange}
                            type="password"
                            placeholder="Password"
                        />
                        <button>Login</button>
                    </form>

                </div>
            );
        }
    }
}