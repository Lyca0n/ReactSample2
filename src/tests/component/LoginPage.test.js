import React from 'react';
import { shallow } from 'enzyme';
import LoginPage from '../../component/ExpenseDashboardPage';
import LoginForm from '../../component/LoginForm';

test('should render LoginPage correctly',()=>{
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});