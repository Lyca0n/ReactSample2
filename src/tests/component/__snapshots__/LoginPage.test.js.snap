// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
<div>
  <h2>
    Login
  </h2>
  <button>
    Login With Google
  </button>
</div>
`;
